alpenhorn/storage.py

```diff
@@ -52,26 +108,129 @@ class StorageNode(base_model):
         How much free space is there on this node?
     avail_gb_last_checked : datetime
         When was the amount of free space last checked?
-    min_delete_age_days : float
-        What is the minimum amount of time a file must remain on the node before
-        we are allowed to delete it?
     notes : string
         Any notes or comments about this node.
     """
 
-    name = pw.CharField(max_length=64)
+    name = pw.CharField(max_length=64, unique=True)
     root = pw.CharField(max_length=255, null=True)
     host = pw.CharField(max_length=64, null=True)
     username = pw.CharField(max_length=64, null=True)
     address = pw.CharField(max_length=255, null=True)
     group = pw.ForeignKeyField(StorageGroup, backref="nodes")
     active = pw.BooleanField(default=False)
     auto_import = pw.BooleanField(default=False)
-    suspect = pw.BooleanField(default=False)
     storage_type = EnumField(["A", "T", "F"], default="A")
-    max_total_gb = pw.FloatField(default=-1.0)
-    min_avail_gb = pw.FloatField()
+    max_total_gb = pw.FloatField(null=True)
+    min_avail_gb = pw.FloatField(default=0)
     avail_gb = pw.FloatField(null=True)
     avail_gb_last_checked = pw.DateTimeField(null=True)
-    min_delete_age_days = pw.FloatField(default=30)
     notes = pw.TextField(null=True)
+
+    @property
+    def local(self) -> bool:
+        """Is this node local to where we are running?"""
+        return self.host == util.get_hostname()
+
+    @property
+    def archive(self) -> bool:
+        """Is this node an archival node?"""
+        return self.storage_type == "A"
+
```

Comment:
Any of the methods below without arguments could probably be properties, for consistency

Reply:
Sure.  I can do that.  `under_min` definitely should be a property.

In Python, is there a concept of "maximum work" before a method should be used instead of a property?  The `all_files` method could be a property but it could potentially generate thousands of `pathlib.Path`s every time it's accessed.  `total_gb` (also called by `over_max`) performs an expensive SQL query.  Are either of those a problem?

Reply:
putting a verb into the name of a method isn't a bad idea.